powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/persistence/ActivationQueryServiceDefault.java

```diff
@@ -0,0 +1,132 @@
+        } catch (Exception ex) {
+            logger.error("Activation query failed", ex);
+            return Optional.empty();
+        }
+    }
+
+    @Override
+    public Optional<ActivationRecordEntity> findActivationWithoutLock(String activationId) {
+        try {
+            return activationRepository.findActivationWithoutLock(activationId);
+        } catch (Exception ex) {
+            logger.error("Activation query failed", ex);
+            return Optional.empty();
+        }
+    }
+
+    @Override
+    public List<ActivationRecordEntity> findByUserIdAndActivationStatusIn(String userId, Set<ActivationStatus> activationStatuses, Pageable pageable) {
+        try {
+            return activationRepository.findByUserIdAndActivationStatusIn(userId, activationStatuses, pageable);
+        } catch (Exception ex) {
+            logger.error("Activation query failed", ex);
+            return Collections.emptyList();
+        }
+    }
+
+    @Override
+    public List<ActivationRecordEntity> findByApplicationIdAndUserIdAndActivationStatusIn(String applicationId, String userId, Set<ActivationStatus> activationStatuses, Pageable pageable) {
+        try {
+            return activationRepository.findByApplicationIdAndUserIdAndActivationStatusIn(applicationId, userId, activationStatuses, pageable);
+        } catch (Exception ex) {
+            logger.error("Activation query failed", ex);
+            return Collections.emptyList();
+        }
+    }
+
+    @Override
+    public Optional<ActivationRecordEntity> findActivationByCodeWithoutLock(String applicationId, String activationCode, Collection<ActivationStatus> states, Date currentTimestamp) {
+        try {
+            return activationRepository.findActivationByCodeWithoutLock(applicationId, activationCode, states, currentTimestamp);
+        } catch (Exception ex) {
+            logger.error("Activation query failed", ex);
+            return Optional.empty();
+        }
+    }
+
+    @Override
+    public List<ActivationRecordEntity> lookupActivations(Collection<String> userIds, Collection<String> applicationIds, Date timestampLastUsedBefore, Date timestampLastUsedAfter, Collection<ActivationStatus> states) {
+        try {
+            return activationRepository.lookupActivations(userIds, applicationIds, timestampLastUsedBefore, timestampLastUsedAfter, states);
+        } catch (Exception ex) {
+            logger.error("Activation query failed", ex);
+            return Collections.emptyList();
+        }
+    }
+
+    @Override
+    public Stream<ActivationRecordEntity> findAbandonedActivations(Collection<ActivationStatus> states, Date startingTimestamp, Date currentTimestamp) {
+        try {
+            return activationRepository.findAbandonedActivations(states, startingTimestamp, currentTimestamp);
+        } catch (Exception ex) {
+            logger.error("Activation query failed", ex);
+            return Stream.empty();
```

Comment:
🤔 We may replace it by a template method pattern or closure.

Reply:
We can refactor it later.